sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Linq/ExpressionNormalizer.cs

```diff
@@ -0,0 +1,288 @@
+            }
+
+            RecordRewrite(b, visited);
+
+            return visited;
+        }
+
+        internal override Expression VisitUnary(UnaryExpression u)
+        {
+            UnaryExpression visited = (UnaryExpression)base.VisitUnary(u);
+            Expression result = visited;
+
+            RecordRewrite(u, result);
+
+            return result;
+        }
+
+        internal override Expression VisitMethodCall(MethodCallExpression call)
+        {
+            Expression visited = VisitMethodCallNoRewrite(call);
+            RecordRewrite(call, visited);
+            return visited;
+        }
+
+        internal Expression VisitMethodCallNoRewrite(MethodCallExpression call)
+        {
+            MethodCallExpression visited = (MethodCallExpression)base.VisitMethodCall(call);
+
+            if (visited.Method.IsStatic
+                && visited.Method.Name == "Equals"
+                && visited.Arguments.Count > 1)
+            {
+                return Expression.Equal(visited.Arguments[0], visited.Arguments[1], false, visited.Method);
+            }
+
+            if (!visited.Method.IsStatic
+                && visited.Method.Name == "Equals"
+                && visited.Arguments.Count > 0)
+            {
+                return CreateRelationalOperator(ExpressionType.Equal, visited.Object, visited.Arguments[0]);
+            }
+
+            if (visited.Method.IsStatic
+                && visited.Method.Name == "CompareString"
+                && visited.Method.DeclaringType.FullName == "Microsoft.VisualBasic.CompilerServices.Operators")
+            {
+                return CreateCompareExpression(visited.Arguments[0], visited.Arguments[1]);
+            }
+
+            if (!visited.Method.IsStatic
+                && visited.Method.Name == "CompareTo"
+                && visited.Arguments.Count == 1
+                && visited.Method.ReturnType == typeof(int))
+            {
+                return CreateCompareExpression(visited.Object, visited.Arguments[0]);
+            }
+
+            if (visited.Method.IsStatic
+                && visited.Method.Name == "Compare"
+                && visited.Arguments.Count > 1
+                && visited.Method.ReturnType == typeof(int))
+            {
+                return CreateCompareExpression(visited.Arguments[0], visited.Arguments[1]);
+            }
+
+            if (visited.Method.Name == nameof(string.StartsWith) && visited.Method.DeclaringType == typeof(string))
+            {
+                return Expression.Call(typeof(DigitalTwinsFunctions).GetMethod(nameof(DigitalTwinsFunctions.StartsWith)), visited.Object, visited.Arguments[0]);
+            }
+
+            if (visited.Method.Name == nameof(string.EndsWith) && visited.Method.DeclaringType == typeof(string))
+            {
+                return Expression.Call(typeof(DigitalTwinsFunctions).GetMethod(nameof(DigitalTwinsFunctions.EndsWith)), visited.Object, visited.Arguments[0]);
+            }
+
```

Comment:
Nit - add a TODO to normalize other expressions like `.Where(r => r.Occupied is bool)` into `.Where(r => DigitalTwinsFunctions.IsBool(r.Occupied))`.